app/views/new.erb

```diff
@@ -16,48 +16,48 @@
       <div class="col-md-10">
       <span class="label label-default">Birthdate</span>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[birthdate][month]">
+          <input type="text" class="form-control" name="robot[birthdate][month]" maxlength="2" require="true">
         </div>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[birthdate][day]">
+          <input type="text" class="form-control" name="robot[birthdate][day]" maxlength="2" require="true">
         </div>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[birthdate][year]">
+          <input type="text" class="form-control" name="robot[birthdate][year]" minlength="4" maxlength="4" require="true">
         </div>
       </div>
     </div>
     <div class="form-group">
       <div class="col-md-10">
         <span class="label label-default">Date Hired</span>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[date_hired][month]">
+          <input type="text" class="form-control" name="robot[date_hired][month]" maxlength="2" required="true">
         </div>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[date_hired][day]">
+          <input type="text" class="form-control" name="robot[date_hired][day]" maxlength="2" required="true">
         </div>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[date_hired][year]">
+          <input type="text" class="form-control" name="robot[date_hired][year]" minlength="4" maxlength="4" required="true">
         </div>
       </div>
     </div>
     <div class="form-group">
       <div class="col-md-10">
         <div class="radio">
           <label>
-            <input type="radio" name="robot[department]" value="option1">
-            Department 1
+            <input type="radio" name="robot[department]" value="RoboPolice">
```

Comment:
rather than hardcoding these constants (RoboPolice, Librarian, and Trash Compactor), a better approach would be to store them in a constant, perhaps a hash that stores their option number as well? then, we could iterate over that constant here rather than having to type all the HTML three times 👍